pkg/apis/pipeline/v1alpha1/verificationpolicy_validation.go

```diff
@@ -61,8 +61,22 @@ func (vs *VerificationPolicySpec) Validate(ctx context.Context) (errs *apis.Fiel
 // Validate KeyRef will check if one of KeyRef's Data or SecretRef exists, and the
 // Supported HashAlgorithm is in supportedSignatureAlgorithms.
 func (key *KeyRef) Validate(ctx context.Context) (errs *apis.FieldError) {
-	if key.Data == "" && key.SecretRef == nil {
-		errs = errs.Also(apis.ErrMissingOneOf("data", "secretref"))
+	keyCount := 0
+	if key.Data != "" {
+		keyCount++
+	}
+	if key.SecretRef != nil {
+		keyCount++
+	}
+	if key.KMS != "" {
+		keyCount++
+	}
+
+	if keyCount == 0 {
+		errs = errs.Also(apis.ErrMissingOneOf("data", "kms", "secretref"))
+	}
+	if keyCount > 1 {
+		errs = errs.Also(apis.ErrMultipleOneOf("data", "kms", "secretref"))
 	}
 
 	if key.Data != "" && key.SecretRef != nil {
```

Comment:
Since multiple keys are checked via keyCount, `if key.Data != "" && key.SecretRef != nil {` seems not needed anymore.

Reply:
thanks I missed this one